### ISSUE TITLE
Build failure: kabeljau (on Darwin)

### ISSUE THREAD

**Issue by sarahec** [NixOS/nixpkgs#343346]
### Steps To Reproduce

1. build `kabeljau` on Darwin

### Build log

```
Running phase: installPhase
+ test -f /nix/store/z9655l0m7bxz5g37xl01yv70lf2md1dl-kabeljau-1.2.0/bin/kabeljau
+ test -f /nix/store/z9655l0m7bxz5g37xl01yv70lf2md1dl-kabeljau-1.2.0/share/applications/kabeljau.desktop
+ for icon_width in 32 64 128
+ test -f /nix/store/z9655l0m7bxz5g37xl01yv70lf2md1dl-kabeljau-1.2.0/share/icons/hicolor/32x32/apps/kabeljau.png
+ for icon_width in 32 64 128
+ test -f /nix/store/z9655l0m7bxz5g37xl01yv70lf2md1dl-kabeljau-1.2.0/share/icons/hicolor/64x64/apps/kabeljau.png
+ for icon_width in 32 64 128
+ test -f /nix/store/z9655l0m7bxz5g37xl01yv70lf2md1dl-kabeljau-1.2.0/share/icons/hicolor/128x128/apps/kabeljau.png
+ mkdir -p ./build
+ rm -rf ./build
+ install -Dm700 kabeljau.sh ./build/bin/kabeljau
+ install -Dm600 kabeljau.desktop -t ./build/usr/share/applications
+ mkdir -p ./build/usr/share/icons/hicolor
+ for icon_width in 32 64 128
+ mkdir -p ./build/usr/share/icons/hicolor/32x32/apps
+ inkscape -o ./build/usr/share/icons/hicolor/32x32/apps/kabeljau.png -C -w 32 -h 32 --export-png-color-mode=RGBA_8 icons/hicolor/32x32/kabeljau.svg

** (process:91057): WARNING **: 20:22:59.393: Unable to create profile directory (Read-only file system) (30)
** Message: 20:22:59.679: Cannot create profile directory /homeless-shelter/.config/inkscape.
** Message: 20:22:59.679: Inkscape will run with default settings, and new settings will not be saved. 
Fontconfig error: Cannot load default config file: No such file: (null)
Fontconfig error: No writable cache directories
Fontconfig error: No writable cache directories
Fontconfig error: No writable cache directories
Fontconfig error: No writable cache directories
Fontconfig error: No writable cache directories
Fontconfig error: No writable cache directories

** (org.inkscape.Inkscape:91057): WARNING **: 20:22:59.770: Could not create directory '/homeless-shelter/.config/inkscape'

** (org.inkscape.Inkscape:91057): WARNING **: 20:22:59.770: Could not create extension error log file '/homeless-shelter/.config/inkscape/extension-errors.log'

(org.inkscape.Inkscape:91057): Gtk-CRITICAL **: 20:22:59.793: Unable to create user data directory '/homeless-shelter/.local/share' for storing the recently used files list: Read-only file system
2024-09-20 20:22:59.851 .inkscape-wrapped[91057:2519730] +[NSXPCSharedListener endpointForReply:withListenerName:replyErrorCode:]: an error occurred while attempting to obtain endpoint for listener 'ClientCallsAuxiliary': Connection interrupted
/private/tmp/nix-build-kabeljau-1.2.0.drv-0/just-wdz7Um/build: line 21: 91057 Trace/BPT trap: 5       inkscape -o ./build/usr/share/icons/hicolor/$icon_width"x"$icon_width/apps/kabeljau.png -C -w $icon_width -h $icon_width --export-png-color-mode=RGBA_8 icons/hicolor/32x32/kabeljau.svg
error: Recipe `build` failed with exit code 133
error: builder for '/nix/store/m1c00pjvxdky1mzpy9h6ik943jpl4q5a-kabeljau-1.2.0.drv' failed with exit code 133;
       last 10 log lines:
       > Fontconfig error: No writable cache directories
       >
       > ** (org.inkscape.Inkscape:91057): WARNING **: 20:22:59.770: Could not create directory '/homeless-shelter/.config/inkscape'
       >
       > ** (org.inkscape.Inkscape:91057): WARNING **: 20:22:59.770: Could not create extension error log file '/homeless-shelter/.config/inkscape/extension-errors.log'
       >
       > (org.inkscape.Inkscape:91057): Gtk-CRITICAL **: 20:22:59.793: Unable to create user data directory '/homeless-shelter/.local/share' for storing the recently used files list: Read-only file system
       > 2024-09-20 20:22:59.851 .inkscape-wrapped[91057:2519730] +[NSXPCSharedListener endpointForReply:withListenerName:replyErrorCode:]: an error occurred while attempting to obtain endpoint for listener 'ClientCallsAuxiliary': Connection interrupted
       > /private/tmp/nix-build-kabeljau-1.2.0.drv-0/just-wdz7Um/build: line 21: 91057 Trace/BPT trap: 5       inkscape -o ./build/usr/share/icons/hicolor/$icon_width"x"$icon_width/apps/kabeljau.png -C -w $icon_width -h $icon_width --export-png-color-mode=RGBA_8 icons/hicolor/32x32/kabeljau.svg
       > error: Recipe `build` failed with exit code 133
```

### Additional context

Calls Inkscape which tries to create working files in the sandbox.

### Notify maintainers

@auroraanna 

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 24.0.0, macOS 15.0`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.18.5`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by auroraanna** [comment]
Is this really an Inkscape problem and not a this directory can't be created on darwin problem?

**Comment by sarahec** [comment]
> Is this really an Inkscape problem and not a this directory can't be created on darwin problem?

It's "inkscape breaks when run in the sandbox".

I wonder if a tempdir would solve it.

**Comment by sarahec** [comment]
Here's the relevant bit of the log:

```sh
2024-09-20 20:22:59.851 .inkscape-wrapped[91057:2519730] +[NSXPCSharedListener endpointForReply:withListenerName:replyErrorCode:]: an error occurred while attempting to obtain endpoint for listener 'ClientCallsAuxiliary': Connection interrupted
/private/tmp/nix-build-kabeljau-1.2.0.drv-0/just-wdz7Um/build: line 21: 91057 Trace/BPT trap: 5       inkscape -o ./build/usr/share/icons/hicolor/$icon_width"x"$icon_width/apps/kabeljau.png -C -w $icon_width -h $icon_width --export-png-color-mode=RGBA_8 icons/hicolor/32x32/kabeljau.svg
```

`Trace/BPT Trap 5` means Inkscape crashed. 

Unfortunately, this winds up hanging `nixpkgs-review` that's used after a change. I have a patch on hold while we clear these blockages (the patch unblocks 100+ packages on Darwin).

Ii might work if inkscape was launched into a writable temp directory.

**Comment by auroraanna** [comment]
i can't test this since i don't have a darwin machine…

**Comment by auroraanna** [comment]
i guess the justfile can't be modified because it'd be a nix-specific solution

**Comment by sarahec** [comment]
Closing per @viraptor (couldn't reproduce)

**Comment by sarahec** [comment]
> i guess the justfile can't be modified upstream because it'd be a nix-specific solution

`cairosvg` is cross-platform so it would work everywhere. It doesn't handle fancy text layouts well, but if you don't have that it would be a safe solution.

**Comment by sarahec** [comment]
Please see #343629n for a working solution using `cairosvg`
